File: Algorithm/Baekjoon/Python/기본 수학2/골드바흐의 추측.py
def check(n):
    if n == 1:
        return False
    d = int(n**0.5)
    for i in range(2, d + 1):
        if n % i == 0:
            return False
    return True


def adder(a, b):
    if check(a) and check(b):
        return a
    else:
        return adder(a - 1, b + 1)


tc = int(input())
for i in range(tc):
    n = int(input())
    if check(n / 2):
        print(str(int(n / 2)) + " " + str(int(n / 2)))
    else:
        j = adder(n / 2, n / 2)
        print(str(int(j)) + " " + str(int(n - j)))

File: Algorithm/Baekjoon/Java/기본_수학_2/골드바흐의_추측.java
package Algorithm.Baekjoon.Java.기본_수학_2;

import java.util.*;

public class 골드바흐의_추측 {
    static boolean check(int n) {
        if (n == 1) {
            return false;
        }
        int tmp = (int) Math.sqrt(n) + 1;
        for (int i = 2; i < tmp; i++) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }

    static int adder(int a, int b) {
        if (check(a) && check(b)) {
            return a;
        }
        return adder(a - 1, b + 1);

    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int tc = sc.nextInt();
        for (int t = 0; t < tc; t++) {
            int n = sc.nextInt();
            if (check(n / 2)) {
                System.out.println(n / 2 + " " + n / 2);
            } else {
                int tmp = adder(n / 2, n / 2);
                System.out.println(tmp + " " + (n - tmp));
            }
        }

        sc.close();
    }
}
